entity/templates/src/main/java/package/service/impl/_EntityServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package <%=packageName%>.service<% if (service == 'serviceImpl') { %>.impl<% } %>;
+<%  var serviceClassName = service == 'serviceImpl' ? entityClass + 'ServiceImpl' : entityClass + 'Service';
+    var viaService = false;
+    var instanceType = (dto == 'mapstruct') ? entityClass + 'DTO' : entityClass;
+    var instanceName = (dto == 'mapstruct') ? entityInstance + 'DTO' : entityInstance;
+    var mapper = entityInstance  + 'Mapper';
+    var dtoToEntity = mapper + '.'+ entityInstance +'DTOTo' + entityClass;
+    var entityToDto = mapper + '.'+ entityInstance +'To' + entityClass + 'DTO';
+    var entityToDtoReference = mapper + '::'+ entityInstance +'To' + entityClass + 'DTO';
+    var repository = entityInstance  + 'Repository';
+    var searchRepository = entityInstance  + 'SearchRepository';
+    if (service == 'serviceImpl') { %>
+import <%=packageName%>.service.<%= entityClass %>Service;<% } %>
+import <%=packageName%>.domain.<%= entityClass %>;
+import <%=packageName%>.repository.<%= entityClass %>Repository;<% if (searchEngine == 'elasticsearch') { %>
+import <%=packageName%>.repository.search.<%= entityClass %>SearchRepository;<% } if (dto == 'mapstruct') { %>
+import <%=packageName%>.web.rest.dto.<%= entityClass %>DTO;
+import <%=packageName%>.web.rest.mapper.<%= entityClass %>Mapper;<% } %>
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;<% if (pagination != 'no') { %>
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;<% } if (databaseType == 'sql') { %>
+import org.springframework.transaction.annotation.Transactional;<% } %>
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;<% if (dto == 'mapstruct') { %>
+import java.util.LinkedList;<% } %>
+import java.util.List;
+import java.util.Optional;<% if (databaseType == 'cassandra') { %>
+import java.util.UUID;<% } %><% if (searchEngine == 'elasticsearch' || dto == 'mapstruct' || fieldsContainNoOwnerOneToOne == true) { %>
+import java.util.stream.Collectors;<% } %><% if (searchEngine == 'elasticsearch' || fieldsContainNoOwnerOneToOne == true) { %>
+import java.util.stream.StreamSupport;<% } %><% if (searchEngine == 'elasticsearch') { %>
+
+import static org.elasticsearch.index.query.QueryBuilders.*;<% } %>
+
+/**
+ * Service Implementation for managing <%= entityClass %>.
+ */
+@Service
+public class <%= serviceClassName %> <% if (service == 'serviceImpl') { %>implements <%= entityClass %>Service<% } %>{
+
+    private final Logger log = LoggerFactory.getLogger(<%= serviceClassName %>.class);
+    <%- include('../../common/inject_template', {viaService: viaService}); -%>
+    /**
+     * Save a <%= entityInstance %>.
+     * @param input the <%= instanceType %>
+     * @return the persisted entity
+     */<% if (databaseType == 'sql') { %>
+    @Transactional(readOnly = false) <% } %>
```

Comment:
It should be transactional at the class level, that's for sure, then the readonly flag is only an optimization when you are not writing data -> so for me we should have @Transactional at the class level and @Transactional(readonly=true) for some methods. The readonly flag can also be forgotten, that's not a big deal, so for most people this configuration will work all the time, even if they don't understand what the flag does -> if we put it on top, that will surprise a lot of people, because if you didn't see the flag at the top of the class (or don't understand it), you'll get lots of errors.